src/zope/i18nmessageid/tests.py

```diff
@@ -15,146 +15,202 @@
 """
 import sys
 import unittest
-
 from zope.i18nmessageid import message as messageid
 
+
 class PyMessageTests(unittest.TestCase):
 
-    _TEST_REAOONLY = True
+    _TEST_READONLY = True
 
     def _getTargetClass(self):
         return messageid.pyMessage
 
     def _makeOne(self, *args, **kw):
         return self._getTargetClass()(*args, **kw)
 
-    def test_ctor_defaults(self):
+    def test_defaults(self):
         message = self._makeOne('testing')
         self.assertEqual(message, 'testing')
         self.assertEqual(message.domain, None)
         self.assertEqual(message.default, None)
         self.assertEqual(message.mapping, None)
-        if self._TEST_REAOONLY:
+        self.assertEqual(message.msgid_plural, None)
+        self.assertEqual(message.default_plural, None)
+        self.assertEqual(message.number, None)
+        if self._TEST_READONLY:
+            self.assertTrue(message._readonly)
+
+    def test_values(self):
+        mapping = {'key': 'value'}
+        message = self._makeOne(
+            'testing', 'domain', 'default', mapping,
+            msgid_plural='testings', default_plural="defaults", number=2)
+        self.assertEqual(message, 'testing')
+        self.assertEqual(message.domain, 'domain')
+        self.assertEqual(message.default, 'default')
+        self.assertEqual(message.mapping, mapping)
+        self.assertEqual(message.msgid_plural, 'testings')
+        self.assertEqual(message.default_plural, 'defaults')
+        self.assertEqual(message.number, 2)
+        if self._TEST_READONLY:
             self.assertTrue(message._readonly)
 
-    def test_ctor_explicit(self):
+    def test_values_with_float_for_number(self):
         mapping = {'key': 'value'}
-        message = self._makeOne('testing', 'domain', 'default', mapping)
+        message = self._makeOne(
+            'testing', 'domain', 'default', mapping,
+            msgid_plural='testings', default_plural="defaults", number=2.2)
         self.assertEqual(message, 'testing')
         self.assertEqual(message.domain, 'domain')
         self.assertEqual(message.default, 'default')
         self.assertEqual(message.mapping, mapping)
-        if self._TEST_REAOONLY:
+        self.assertEqual(message.msgid_plural, 'testings')
+        self.assertEqual(message.default_plural, 'defaults')
+        self.assertEqual(message.number, 2.2)
+        if self._TEST_READONLY:
             self.assertTrue(message._readonly)
 
-    def test_ctor_copy(self):
+    def test_copy(self):
         mapping = {'key': 'value'}
-        source = self._makeOne('testing', 'domain', 'default', mapping)
+        source = self._makeOne(
+            'testing', 'domain', 'default', mapping,
+            msgid_plural='testings', default_plural="defaults", number=2)
```

Comment:
Change this to `number=0` and you'll see if I'm right about the bug in the code (I expect message.number to end up being `None`).